### ISSUE TITLE
PoeCdn image hex encoding

### ISSUE THREAD

**Issue by Kvan7** [Kvan7/Exiled-Exchange-2#30]
Images stored are available on the cdn:
https://web.poecdn.com/gen/image/WzI1LDE0LHsiZiI6IjJESXRlbXMvV2VhcG9ucy9Ud29IYW5kV2VhcG9ucy9Cb3dzL1VuaXF1ZXMvRGVhdGhzSGFycCIsInciOjIsImgiOjQsInNjYWxlIjoxLCJyZWFsbSI6InBvZTIifV0/2205f4c8b2/DeathsHarp.png
I've figured out most of this but i need to try and figure out the hex part `2205f4c8b2`. Going to wait on this since the images aren't? strictly nessesary

**Comment by Kvan7** [comment]
May mark this as blocking, I need to see how bad it is without images. 

Normally these are pulled from the trade api, and currently i am kind of hacking it together without hitting the trade api.
Either need to decode the hex or write something to hit the trade site

**Comment by maksim789456** [comment]
This part is some kind of custom made hash function, where maybe base64 part is input, to prevent URL creating and CDN/Cache 'busting'. I don't think there are ways to generate this hash

> Based on ['zao' message](https://discord.com/channels/174993814845521922/175290321695932416/1231574561246416926) in #tooldev-general on GGG discord
> ![image](https://github.com/user-attachments/assets/1c5f87a0-7b0d-46b0-9b02-293c06e55f4c)

**Comment by maksim789456** [comment]
I'm also tried to hashing the base64 part, decoded base64 part, file path and other values by some popular hash functions (like base64, md5, sha1 and etc.) and also fnv1a64, MurmurHash64A hash functions, witch be used as hashing function for paths in index at GGPK file, but all results not similar with value in URL

**Comment by Kvan7** [comment]
Yeah, right now I have hacked in some, just copying old cdn urls from the old items into the new ones. 
Looks like I can easily get the currency item images, but all items will take a bit more work.

**Comment by rwygand** [comment]
Where is this used? I'd assume that if the poecdn is serving up images with cache-busting hashed that cannot be calculated by clients, then there must be some way to resolve an item to an image URL offered either in the game files or the poe apis?

**Comment by Kvan7** [comment]
This is used just to show the items with the overlays pop up. It's not that big a deal, and i added an image not found image so this is fine waiting until the api is open.

**Comment by maksim789456** [comment]
@Kvan7 as temp variant you can use items image from poe2db CDN
https://cdn.poe2db.tw/image/art/2ditems/weapons/twohandweapons/bows/uniques/deathsharp.webp
URL just contains path to image as in GGPK, it's easy to generate
